cl_sii/rut/crypto_utils.py

```diff
@@ -0,0 +1,47 @@
+import re
+
+import cryptography
+import cryptography.x509
+from cryptography.hazmat.backends.openssl import backend as crypto_x509_backend
+
+from . import constants
+from . import Rut
+
+
+def get_subject_rut_from_certificate_pfx(pfx_file_bytes: bytes, password: str) -> Rut:
+    """
+    Return the Chilean RUT stored in a digital certificate.
+
+    :param pfx_file_bytes: Digital certificate in PKCS12 format
+    :param password: (Optional) The password to use to decrypt the PKCS12 file
+    """
+    private_key, x509_cert, additional_certs = crypto_x509_backend.load_key_and_certificates_from_pkcs12(  # noqa: E501
+        data=pfx_file_bytes,
+        password=password.encode(),
+    )
+    # https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization/#cryptography.hazmat.primitives.serialization.pkcs12.load_key_and_certificates  # noqa: E501
+
+    subject_alt_name_ext = x509_cert.extensions.get_extension_for_class(
+        cryptography.x509.extensions.SubjectAlternativeName,
+    )
+
+    # Search for the RUT in the certificate.
+    try:
+        results = [
+            x.value
+            for x in subject_alt_name_ext.value._general_names
+            if x.type_id == constants.SII_CERT_TITULAR_RUT_OID
+        ]
+    except AttributeError:
+        raise Exception('Certificate has no RUT information')
+
+    if not results:
+        raise Exception('Certificate has no RUT information')
+    elif len(results) > 1:
+        raise Exception(f'len(results) == {len(results)}')
+
+    subject_rut_raw: bytes = results[0]
+    subject_rut = re.sub(r'[^0-9-]', '', subject_rut_raw.decode('utf-8'))
+    subject_rut = subject_rut.lstrip('0')  # remove leading zeroes
```

Comment:
I think that removing leading zeroes isn't necessary anymore as `Rut` already does it, and it's _theoretically_ possible for `subject_rut` to be `0-0`, which `lstrip` would transform to `-0`.

Reply:
Nice catch, I completely forgot about the PR https://github.com/fyntex/lib-cl-sii-python/pull/238